pkg/controller/spinnakervalidating/spinnakervalidating_controller.go

```diff
@@ -30,19 +38,129 @@ var log = logf.Log.WithName("spinvalidate")
 
 // Add adds the validating admission controller
 func Add(m manager.Manager) error {
-	//ns, err := k8sutil.GetOperatorNamespace()
-	_, err := k8sutil.GetOperatorNamespace()
+	// Determine environment
+	ns, name, err := getOperatorNameAndNamespace()
 	if err != nil {
 		return err
 	}
 
+	// Create Kubernetes service for listening to requests from API server
+	rawClient := kubernetes.NewForConfigOrDie(m.GetConfig())
+	port := 9876
+	err = deployWebhookService(ns, name, port, rawClient)
+	if err != nil {
+		return err
+	}
+
+	// Create or get certificates
+	c, err := getCertContext(ns, name)
+	if err != nil {
+		return err
+	}
+
+	// Register webhook server
 	hookServer := m.GetWebhookServer()
-	hookServer.Register("/validate-v1alpha1-spinnakerservice", &webhook.Admission{Handler: &spinnakerValidatingController{}})
-	return nil
+	hookServer.CertDir = c.certDir
+	hookServer.Port = port
+	gv := SpinnakerServiceBuilder.GetGroupVersion()
+	path := fmt.Sprintf("/validate-%s-spinnakerservice", gv.Version)
+	hookConfigName := fmt.Sprintf("spinnakervalidatingwebhook.%s", gv.Group)
+	hookServer.Register(path, &webhook.Admission{Handler: &spinnakerValidatingController{}})
+
+	// Create validating webhook configuration for registering our webhook with the API server
+	w := getWebhookConfig(hookConfigName, name, ns, path, c)
+	return deployValidatingWebhookConfiguration(hookConfigName, ns, w, rawClient)
+}
+
+func getOperatorNameAndNamespace() (string, string, error) {
+	name, err := k8sutil.GetOperatorName()
+	if err != nil {
+		return "", "", err
+	}
+	ns, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		envNs := os.Getenv("ADMISSION_PROXY_NAMESPACE")
+		if envNs == "" {
+			return "", "", fmt.Errorf("unable to determine operator namespace. Error: %s and ADMISSION_PROXY_NAMESPACE env var not set", err.Error())
+		}
+		ns = envNs
+	}
+	return ns, name, nil
 }
 
-// spinnakerValidatingController adds an annotation to every incoming pods.
+func deployWebhookService(ns string, name string, port int, client *kubernetes.Clientset) error {
+	// Always recreate the service
```

Comment:
We should update instead of recreating. Also using the manager's client would be preferable to the raw client.